documentation/src/main/asciidoc/topics/con_performance_data_consistency.adoc

```diff
@@ -0,0 +1,105 @@
+[id='performance-data-consistency_{context}']
+= Data consistency
+
+Data that resides on a distributed system is vulnerable to errors that can arise from temporary network outages, system failures, or just simple human error.
+These external factors are uncontrollable but can have serious consequences for quality of your data.
+The effects of data corruption range from lower customer satisfaction to costly system reconciliation that results in service unavailability.
+
+{brandname} can carry out ACID (atomicity, consistency, isolation, durability) transactions to ensure the cache state is consistent.
+
+Transactions are a sequence of atomic operations that {brandname} caries out as a single operation.
+Either all reads and writes in the transaction complete successfully or they all fail.
+In this way, the transaction either modifies the cache state in a consistent way, providing a history of reads and writes, or it does not modify cache state at all.
+
+The following is a simple transaction with one read and one write operation:
+
+[source,java,options="nowrap",subs=attributes+]
+----
+transaction.begin()
+cache.get(k1) // reads v1
+cache.put(k1, v2) // writes v2
+transaction.commit()
+----
+
+Because each one is composed of a series of operations, transactions increase the contention of read and write operations compared to non-transactional caches.
+
+The main performance concern for enabling transactions is finding the balance between having a more consistent data set and increasing latency that degrades write throughput.
+
+.Write locking
+
+To perform write operations while carrying out transactions, {brandname} locks keys.
+Locking keys adds contention that increases latency for write operations.
+How much contention depends on whether {brandname} locks keys optimistically or pessimistically.
+
+* Optimistic locking
++
+For low rates of contention, {brandname} locks keys when transactions invoke the `commit()` method.
```

Comment:
I would probably replace "rates" with workload: "For low/high contention workloads"